packages/shopify-app-session-storage-prisma/src/prisma.ts

```diff
@@ -38,11 +41,29 @@ export class PrismaSessionStorage<T extends PrismaClient>
 
     const data = this.sessionToRow(session);
 
-    await this.getSessionTable().upsert({
-      where: {id: session.id},
-      update: data,
-      create: data,
-    });
+    try {
+      await this.getSessionTable().upsert({
+        where: {id: session.id},
+        update: data,
+        create: data,
+      });
+    } catch (error) {
+      if (
+        error instanceof Prisma.PrismaClientKnownRequestError &&
+        error.code === UNIQUE_KEY_CONSTRAINT_ERROR_CODE
+      ) {
+        await this.getSessionTable().upsert({
+          where: {id: session.id},
+          update: data,
+          create: data,
+        });
+        console.log(
```

Comment:
Totally nit-picking, but we should probably log before the retry, just in case it blows up somehow 😄